### DISCUSSION TITLE
Openblas slow on sandybridge?

### DISCUSSION THREAD

**Discussion by nardis-miles** [OpenMathLib/OpenBLAS#4962] [Q&A]
I'm rebuilding on two clusters, both running RH Linux: 
1. sandybridge 
Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz
RH-7

2. haswell
Intel(R) Xeon(R) CPU E5-2697 v3 @ 2.60GHz
RH-8

On 2, OpenBLAS gives a consistent speedup of ~2 over an ATLAS application

On 1, however, OpenBLAS is roughly 3x slower. I am mystified. On both, I'm running single-threaded, as I'm using MPI and each core is an MPI instance. Am I missing something?

Art Edwards

**Comment by martin-frbg** [comment]
The old AVX-only Sandybridge architecture has not received much attention lately, but I would not expect OpenBLAS to be slower than ATLAS.
Which version are you building, what compiler and build options did you use, and what BLAS functions does your application primarily use (if known) ?

**Reply by nardis-miles** [reply]
A significant part of the calculation is an eigensolve (Hermitian), and anothera is M-V and MM multiplies for complex double precision. I should note that the LA is inside scalapack that we hand roll. The latter is about 3x as long as the eigensolve for large problems. I should also note that there are some special features of the code. we use a lot of disk reading and writing to conserve memory. As I look at the system, I've noted that, while we use to have local disk storage, it seems that now we are attached to a disk farm. This could be a killing situation. I'll get back to this as soon as I have clarification from the sys admins rebuilding the system.
